node/src/chain_spec/mod.rs

```diff
@@ -42,46 +42,53 @@ use zeitgeist_primitives::{
     constants::BalanceFractionalDecimals,
     types::{AccountId, Balance},
 };
-#[cfg(feature = "parachain")]
-use {
-    sp_runtime::Perbill,
-    zeitgeist_primitives::constants::{ztg, MILLISECS_PER_BLOCK},
-    zeitgeist_runtime::DefaultBlocksPerRound,
-};
 
 cfg_if::cfg_if! {
     if #[cfg(feature = "parachain")] {
         // Common
-        pub(crate) const DEFAULT_COLLATOR_INFLATION_INFO: pallet_parachain_staking::InflationInfo<Balance> = {
-            let hours_per_year = 8766;
-            let millisecs_per_year = hours_per_year * 60 * 60 * 1000;
-            let round_millisecs = DefaultBlocksPerRound::get() as u64 * MILLISECS_PER_BLOCK as u64;
-            let rounds_per_year = millisecs_per_year / round_millisecs;
-
-            let annual_inflation = ztg::STAKING_PTD;
-            let expected_annual_amount = ztg::COLLATORS * zeitgeist_primitives::constants::BASE;
-            let round_inflation_parts = annual_inflation.deconstruct() as u64 / rounds_per_year;
-            let round_inflation = Perbill::from_parts(round_inflation_parts as _);
```

Comment:
I think here use of from_parts was correct as there was deconstruct() method call above.

Reply:
Yes, as far as I can tell the code derived correct per round inflation. However, it was inflexible because it did not allow to use different annual and per round inflation based on different expected staking values. In addition to that, every network had to use the same `InflationConfig` as defined in `mod.rs`. Now we can derive a function to create `InflationConfig` that respects the configuration of the runtime in which it is used.